pkg/network/ndmz/ndmz.go

```diff
@@ -111,23 +147,43 @@ func Delete() error {
 }
 
 func createPublicIface(netNS ns.NetNS) error {
+	var pubIface string
 	if !ifaceutil.Exists(types.PublicIface, netNS) {
 
-		var (
-			master netlink.Link
-			err    error
-		)
-
+		// find which interface to use as master for the macvlan
 		if namespace.Exists(types.PublicNamespace) {
-			master, err = getPublicIface()
+			pubNS, err := namespace.GetByName(types.PublicNamespace)
```

Comment:
this err is not checked

Reply:
well yea.. it Exists(), so could as well be _

Reply:
Not sure about that, the implementation of GetByName seems to also open a file, which could error even if it exists, as far as I know.